### PR TITLE
Feature/add-ge-le

### PR DESCRIPTION
**Pull request by iskandersierra**
Add (greater|less)Than[OrEqualTo]... rules and operations

### CODE REVIEW

**Comment by pimbrouwers** [comment]
Hi! Thanks for your effort! Isn't this covered by between already?

**Comment by iskandersierra** [comment]
Not really. For example, I want to do 
```fsharp
let MinValue = DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)
Check.DateTime.greaterThanOrEqualTo MinValue field value
```
Now I have to do 
```fsharp
let MinValue = DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)
let private GreaterThanMinValue = MinValue.AddMilliseconds(-1)
Check.DateTime.greaterThan GreaterThanMinValue field value
```
If I use `between MinValue DateTime.MaxValue` the message is very weird because the second date is not business related.

**Comment by pimbrouwers** [comment]
This can be represented by the current functions as well, as you've pointed out (but shortened a bit).

```fsharp
let minValue = DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc).AddTicks(-1)
Check.DateTime.greaterThan MinValue field value
```

You can override the default message very easily using `Check.WithMessage.{type}`.

```fsharp
let minValue = DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)
let maxValue = DateTime.MaxValue
let msg field = $"{field} must be greater than or equal to 1970-01-01"
Check.WithMessage.DateTime.between MinValue maxValue msg field value
```

I am not trying to stonewall you, just attempting to point out that I did consider this situation when designing the library.

**Comment by iskandersierra** [comment]
That is great! I can use that solution with my use case. Anyways it seemed strange to me not to have the complete set of comparison rules by default.

Feel free to incorporate my changes or not.

I also caught a couple of typos in messages and test names. Those could be breaking changes for people using the default message in tests or translations, so be careful.

Also, I didn't changed the README to add the new rules.

File: src/Validus/Validators.fs [hunk 1]
```diff
@@ -435,7 +541,14 @@ module Validators =
             /// default error message
             member _.greaterThanLen (min : int) (field : string) (input : 'b) =
                 let msg field =
-                    sprintf "'%s' must be greater than %i items" field min
+                    sprintf "'%s' length must be greater than %i items" field min
```
Comment (pimbrouwers) [inline src/Validus/Validators.fs]: I am not a fan of this style. Please revert, and follow the existing.

**Comment by pimbrouwers** [comment]
Thanks for updating the tests and code comments. Really appreciate it. Complete the PR by updating the docs, fixing the message I commented on and we'll merge it in. 

You're not the first person to ask, and I am beginning to think my approach was wrong.

**Comment by iskandersierra** [comment]
Ok, I'll do, thanks.

I don't think your approach was wrong. You are giving all the tools in the library, but having those new operators are more convenient for certain cases, when the developer doesn't want to dig deeper into the library yet.

File: src/Validus/Validators.fs (hunk 1 above)
Comment (iskandersierra) [reply src/Validus/Validators.fs]: Should it say `sprintf "'%s' must have more than %i items" field min`? Or do I leave it as it was before?

Comment (pimbrouwers) [reply src/Validus/Validators.fs]: Go back to what it was before, and match that style with the new validators you're adding.

Comment (iskandersierra) [reply src/Validus/Validators.fs]: I'll have to check all new messages I added.

**Comment by iskandersierra** [comment]
I think I addressed all your questions. Please let me know if I left something out.

**Comment by pimbrouwers** [comment]
Thank you for work on this!

**Comment by pimbrouwers** [comment]
Deployed as part of 4.0.3

**Comment by iskandersierra** [comment]
It was great working with you! I updated the version:

```fsharp
module TaskDueDate =
    let MinValue =
        DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)

    let value (TaskDueDate value) = value

    let create value =
        validate {
            let! value =
                Check.WithMessage.DateTime.greaterThanOrEqualTo // <---
                    MinValue
                    (sprintf "'%s' must be greater than or equal to 1970-01-01")
                    Fields.DUE_DATE
                    value

            return TaskDueDate value
        }
```

**Pull request merged**